csharp/sources/Valkey.Glide/BaseClient.GenericCommands.cs

```diff
@@ -0,0 +1,64 @@
+// Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0
+
+using Valkey.Glide.Commands;
+using Valkey.Glide.Internals;
+
+namespace Valkey.Glide;
+
+public abstract partial class BaseClient : IGenericBaseCommands
+{
+    public async Task<bool> KeyDeleteAsync(ValkeyKey key, CommandFlags flags = CommandFlags.None)
+        => await Command(Request.KeyDeleteAsync(key));
+
+    public async Task<long> KeyDeleteAsync(ValkeyKey[] keys, CommandFlags flags = CommandFlags.None)
+        => await Command(Request.KeyDeleteAsync(keys));
+
+    public async Task<bool> KeyUnlinkAsync(ValkeyKey key, CommandFlags flags = CommandFlags.None)
+        => await Command(Request.KeyUnlinkAsync(key));
+
+    public async Task<long> KeyUnlinkAsync(ValkeyKey[] keys, CommandFlags flags = CommandFlags.None)
+        => await Command(Request.KeyUnlinkAsync(keys));
+
+    public async Task<bool> KeyExistsAsync(ValkeyKey key, CommandFlags flags = CommandFlags.None)
+        => await Command(Request.KeyExistsAsync(key));
+
+    public async Task<long> KeyExistsAsync(ValkeyKey[] keys, CommandFlags flags = CommandFlags.None)
+        => await Command(Request.KeyExistsAsync(keys));
+
+    public async Task<bool> KeyExpireAsync(ValkeyKey key, TimeSpan? expiry, ExpireWhen when = ExpireWhen.Always, CommandFlags flags = CommandFlags.None)
+        => await Command(Request.KeyExpireAsync(key, expiry, when));
+
+    public async Task<bool> KeyExpireAsync(ValkeyKey key, DateTime? expiry, ExpireWhen when = ExpireWhen.Always, CommandFlags flags = CommandFlags.None)
+        => await Command(Request.KeyExpireAsync(key, expiry, when));
+
+    public async Task<TimeSpan?> KeyTimeToLiveAsync(ValkeyKey key, CommandFlags flags = CommandFlags.None)
+        => await Command(Request.KeyTimeToLiveAsync(key));
+
+    public async Task<ValkeyType> KeyTypeAsync(ValkeyKey key, CommandFlags flags = CommandFlags.None)
+        => await Command(Request.KeyTypeAsync(key));
+
+    public async Task<bool> KeyRenameAsync(ValkeyKey key, ValkeyKey newKey, CommandFlags flags = CommandFlags.None)
+        => await Command(Request.KeyRenameAsync(key, newKey));
+
+    public async Task<bool> KeyRenameNXAsync(ValkeyKey key, ValkeyKey newKey, CommandFlags flags = CommandFlags.None)
+        => await Command(Request.KeyRenameNXAsync(key, newKey));
+
+    public async Task<bool> KeyPersistAsync(ValkeyKey key, CommandFlags flags = CommandFlags.None)
+        => await Command(Request.KeyPersistAsync(key));
+
+    public async Task<byte[]?> KeyDumpAsync(ValkeyKey key, CommandFlags flags = CommandFlags.None)
+        => await Command(Request.KeyDumpAsync(key));
+
+    public async Task KeyRestoreAsync(ValkeyKey key, byte[] value, TimeSpan? expiry = null, CommandFlags flags = CommandFlags.None)
```

Comment:
I think restore has more options on server side. Probably we need to copy API to support both

Reply:
Yes, need to create it and follow SER style (naming, types, etc)